lib/index.js

```diff
@@ -234,10 +242,18 @@ internals.implementation = function(server, options) {
      * @returns {Boolean} true. always return true (unless there's an error...)
      */
     response: function(request, h) {
-      if (options.responseFunc && typeof options.responseFunc === 'function') {
+      const responseFunc = options.responseFunc;
+      if (responseFunc && typeof responseFunc === 'function') {
+        if (
+          internals.checkObjectType(responseFunc) === '[object AsyncFunction]'
+        ) {
+          return responseFunc(request, h)
```

Comment:
@ethanrubio this is a _small_ but _useful_ addition for the people who are using Promises. 👍